src/ai/lua/core.cpp

```diff
@@ -785,26 +785,26 @@ static int impl_ai_aspect_get(lua_State* L)
 	}
 	
 	typedef std::vector<std::string> string_list;
-	if(typesafe_aspect<bool>* aspect = try_aspect_as<bool>(iter->second)) {
-		lua_pushboolean(L, aspect->get());
-	} else if(typesafe_aspect<int>* aspect = try_aspect_as<int>(iter->second)) {
-		lua_pushinteger(L, aspect->get());
-	} else if(typesafe_aspect<double>* aspect = try_aspect_as<double>(iter->second)) {
-		lua_pushnumber(L, aspect->get());
-	} else if(typesafe_aspect<config>* aspect = try_aspect_as<config>(iter->second)) {
-		luaW_pushconfig(L, aspect->get());
-	} else if(typesafe_aspect<string_list>* aspect = try_aspect_as<string_list>(iter->second)) {
-		lua_push(L, aspect->get());
-	} else if(typesafe_aspect<terrain_filter>* aspect = try_aspect_as<terrain_filter>(iter->second)) {
+	if(typesafe_aspect<bool>* aspect_as_bool = try_aspect_as<bool>(iter->second)) {
+		lua_pushboolean(L, aspect_as_bool->get());
+	} else if(typesafe_aspect<int>* aspect_as_int = try_aspect_as<int>(iter->second)) {
+		lua_pushinteger(L, aspect_as_int->get());
+	} else if(typesafe_aspect<double>* aspect_as_double = try_aspect_as<double>(iter->second)) {
+		lua_pushnumber(L, aspect_as_double->get());
+	} else if(typesafe_aspect<config>* aspect_as_config = try_aspect_as<config>(iter->second)) {
+		luaW_pushconfig(L, aspect_as_config->get());
+	} else if(typesafe_aspect<string_list>* aspect_as_string_list = try_aspect_as<string_list>(iter->second)) {
+		lua_push(L, aspect_as_string_list->get());
+	} else if(typesafe_aspect<terrain_filter>* aspect_as_terrain_filter = try_aspect_as<terrain_filter>(iter->second)) {
 		std::set<map_location> result;
-		aspect->get().get_locations(result);
+		aspect_as_terrain_filter->get().get_locations(result);
 		lua_push(L, result);
-	} else if(typesafe_aspect<attacks_vector>* aspect = try_aspect_as<attacks_vector>(iter->second)) {
+	} else if(typesafe_aspect<attacks_vector>* aspect_as_attacks_vector = try_aspect_as<attacks_vector>(iter->second)) {
```

Comment:
The point is that 'shorter lifetime' is a programmer thing. The standard says it lives until the end of the entire if statement.

That is why I said each case needs to be evaluated carefully.

In this specific case, it may be more clear to remove the "typesafe_aspect<attacks_vector>*" than to rename "aspect". But just because it's safe here does not make it universally safe.

The point of this warning message was to say "Hey, you might have meant to use the earlier definition, but you just masked it, so this may not do what you might expect based solely on the variable names and ignoring all scoping rules."